apps/design/backend/src/app.ts

```diff
@@ -882,37 +885,46 @@ export function buildUnauthenticatedApi({ logger }: AppContext) {
       if (validationResult.isErr()) {
         return err(validationResult.err());
       }
-      // Decode and process the payload
-      const messageParts = payload.split('//');
-      const [version, header, message] = messageParts;
 
-      if (version !== '1' || header !== 'qr1') {
-        return err('invalid-payload');
-      }
-
-      const [ballotHash, machineId, isLive, secondsSince1970, tallyB64] =
-        message.split('.');
-      const timestampNum = safeParseNumber(secondsSince1970);
-      if (timestampNum.isErr()) {
-        return err('invalid-payload');
-      }
-
-      const signedTimestamp = new Date(timestampNum.ok() * 1000);
-      let tally: CompressedTally;
       try {
-        const tallyJson = Buffer.from(tallyB64, 'base64').toString('ascii');
-        tally = JSON.parse(tallyJson) as CompressedTally;
-        // TODO (#7151): Load the election for the given ballot hash, save the results report.
-      } catch (error) {
+        const {
+          ballotHash,
+          machineId,
+          isLive,
+          signedTimestamp,
+          encodedCompressedTally,
+        } = decodeQuickResultsMessage(payload);
+
+        const electionRecord =
+          await store.getElectionFromBallotHash(ballotHash);
```

Comment:
So if I (accidentally or intentionally) create an election in VxDesign with the same election, even if it belongs to a different organization, there's a chance that it's _my_ election that will get incoming results instead of the intended one. Given the assumption that the production VxDesign will be locked down sufficiently and not allow accounts with the general public, it seems likely that this would only happen by accident. However, I could easily see that happening if someone created a duplicate as a backup or something.

Reply:
See other comment this will gen a new id and create a new ballot hash